src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -61,62 +61,26 @@ internal static IList<CustomAttributeData> GetCustomAttributesInternal(RuntimeTy
             Debug.Assert(target != null);
 
             IList<CustomAttributeData> cad = GetCustomAttributes(target.GetRuntimeModule(), target.MetadataToken);
-
-            Attribute[] a = PseudoCustomAttribute.GetCustomAttributes((RuntimeType)target, (RuntimeType)typeof(object), out int pcaCount);
-
-            if (pcaCount == 0)
-                return cad;
-
-            CustomAttributeData[] pca = new CustomAttributeData[cad.Count + pcaCount];
-            cad.CopyTo(pca, pcaCount);
-            for (int i = 0; i < pcaCount; i++)
-            {
-                pca[i] = new CustomAttributeData(a[i]);
-            }
-
-            return Array.AsReadOnly(pca);
+            PseudoCustomAttribute.GetCustomAttributes((RuntimeType)target, (RuntimeType)typeof(object), out RuntimeType.ListBuilder<Attribute> pcas);
+            return GetCombinedList(cad, ref pcas);
         }
 
         internal static IList<CustomAttributeData> GetCustomAttributesInternal(RuntimeFieldInfo target)
         {
             Debug.Assert(target != null);
 
             IList<CustomAttributeData> cad = GetCustomAttributes(target.GetRuntimeModule(), target.MetadataToken);
-
-            Attribute[] a = PseudoCustomAttribute.GetCustomAttributes((RuntimeFieldInfo)target, (RuntimeType)typeof(object), out int pcaCount);
-
-            if (pcaCount == 0)
-                return cad;
-
-            CustomAttributeData[] pca = new CustomAttributeData[cad.Count + pcaCount];
-            cad.CopyTo(pca, pcaCount);
-            for (int i = 0; i < pcaCount; i++)
-            {
-                pca[i] = new CustomAttributeData(a[i]);
-            }
-
-            return Array.AsReadOnly(pca);
+            RuntimeType.ListBuilder<Attribute> pcas = PseudoCustomAttribute.GetCustomAttributes((RuntimeFieldInfo)target, (RuntimeType)typeof(object));
```

Comment:
All these should use `out` too like the other one.

Reply:
The reason I left these is a great many are just `=> default;` right now, which seems to be a code consistency thing with *always* calling `PseudoCustomAttribute.GetCustomAttributes` even in paths that will never have any. Obviously this is suboptimal...and making them `out` will just make them suboptimal and ugly.

Thoughts on me removing all these and putting a comment where they were? That thing type of (insert MemberInfo overload) will never page pseudo attributes?

Reply:
I tossed up a commit here to show what I meant by the above: https://github.com/NickCraver/coreclr/commit/7ceffcd4f377cc7ee64bb2a0b34c2bfa325dc1e2 - thoughts?

Reply:
It like that change. Less code is good.

Also, it may look slightly better to pass the ListBuilder by `ref` and let the caller initialize it.